src/mcp/tools/core/set_firebase_directory.ts

```diff
@@ -0,0 +1,43 @@
+import { z } from "zod";
+import { tool } from "../../tool.js";
+import { mcpError, toContent } from "../../util.js";
+import { existsSync } from "fs";
+import { join } from "path";
+
+export const set_firebase_directory = tool(
+  {
+    name: "set_firebase_directory",
+    description:
+      "Sets the project directory for the Firebase MCP server to utilize for project detection and authentication. This should be a directory with a `firebase.json` file in it. This information is persisted between sessions.",
+    inputSchema: z.object({
+      dir: z
+        .string()
+        .nullable()
+        .describe(
+          "the absolute path of the directory. set to null to 'unset' the value and fall back to the working directory",
+        ),
+    }),
+    annotations: {
+      title: "Set Firebase Project Directory",
+      idempotentHint: true,
+    },
+  },
+  ({ dir }, { host }) => {
+    if (dir === null) {
+      host.setProjectRoot(null);
+      return Promise.resolve(
+        toContent(
+          `Firebase MCP project directory setting deleted. New project root is: ${host.projectRoot || "unset"}`,
```

Comment:
Seems like this should never get hit from looking at the code, but seems fine to have a fallback